### DISCUSSION TITLE
Most CPU-efficient way to resize/compress image?

### DISCUSSION THREAD

**Discussion by jozseftuska** [dlemstra/Magick.NET#1480] [Help]
Hi.

We have images stored on blob storage. There are some clients that have slow internet connection so we want to compress images on the fly to create thumbnails. It is working fine quality and size-wise, 3MB files get compressed to 10-20Kb. The only problem is that it uses too much CPU resource and takes sometimes 1-3 seconds.

Our current code:
```csharp
            var data = new byte[originalBlob.Properties.Length];
            await originalBlob.DownloadToByteArrayAsync(data, 0);

            using var image = new MagickImage(data);
            var size = new MagickGeometry(options.Width, options.Height)
            {
                IgnoreAspectRatio = options.IgnoreRatio
            };

            image.Resize(size);
            var imageBytes = image.ToByteArray();

            await resizedFile.UploadFromByteArrayAsync(imageBytes, 0, imageBytes.Length);
```
We could allow worse image quality and bigger size. I've also tried thumbnail but it was still slow.
Could you help how to optimize this code or an alternative approach resizing or even compressing? We would like to keep it under 500ms. 

Thank you for your help in advance.

**Comment by dlemstra** [comment]
The `Thumbnail` method is optimized for speed and removes image metadata that reduces the size so I would advise you to use that. The duration is probably related to the dimensions of your image. The size of the file doesn't matter, the width and height matter. When your image has huge dimensions it can take a while before this is loaded and resized. It might also be faster if you use the Q8-OpenMP version.
